core/cmd/shell_local.go

```diff
@@ -469,14 +469,12 @@ func (s *Shell) runNode(c *cli.Context) error {
 		}
 	}
 
-	if s.Config.Capabilities().WorkflowRegistry().Address() != "" {
-		err2 := app.GetKeyStore().Workflow().EnsureKey(rootCtx)
-		if err2 != nil {
-			return errors.Wrap(err2, "failed to ensure workflow key")
-		}
+	err2 := app.GetKeyStore().Workflow().EnsureKey(rootCtx)
+	if err2 != nil {
+		return errors.Wrap(err2, "failed to ensure workflow key")
```

Comment:
💭 This check was created because we said not all the node would have this key, wouldn't this assume the opposite?

Reply:
Yeah that's right! The problem is that in order to setup the WorkflowRegistry we need a key, and the check will only create one if there is a WorkflowRegistry set; the current workaround is to set it to a dummy address ("0x0") but that would still require two restarts

Reply:
ok got it, thanks 👍🏼